src/bin/nydus-image/builder/tarball.rs

```diff
@@ -543,11 +559,31 @@ impl Builder for TarballBuilder {
                     &mut bootstrap_ctx,
                     &mut bootstrap,
                     blob_mgr,
-                    &mut writer,
+                    bootstrap_writer,
                 )
             },
             "dump_bootstrap"
         )?;
+
+        if let Some((_, blob_ctx)) = blob_mgr.get_current_blob() {
```

Comment:
Does the directory build need to write TOC too?

Reply:
Fixed.